src/component.ts

```diff
@@ -68,12 +72,24 @@ export function extendWithHostListeners(ctrl: {new(...args: any[])}, listeners:
         const { eventName } = listeners[handler];
         this.$element.on(eventName + namespace, this[handler].bind(this));
       });
+      properties.forEach(property => {
+        const query = queries[property];
+        const elements = this.$element[0].querySelector(query.selector);
+        if (query.first) {
+          this[property] = angular.element(elements[0]);
+        }
+        else {
+          this[property] = angular.element(elements);
```

Comment:
```
const viewChildEls = this.$element[0].querySelectorAll(query.selector);
this[property] = viewChildEls.map(viewChild => angular.element(viewChild).isolateScope<any>()['$ctrl'])
```

Reply:
It's possible to pass child component's type to the decorator and get it's selector from it's name when you define view children , and then you can get to $ctrl property on isolateScope() method of angular.element, that is viewChild controller.
Left you comments in code that need to be changed.
In general I would rename all the "query" names to "viewChild"